Scripts/Modules/SnappingModule/SnappingModule.cs

```diff
@@ -0,0 +1,709 @@
+﻿#if UNITY_EDITOR
+using System.Collections.Generic;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+
+namespace UnityEditor.Experimental.EditorVR.Modules
+{
+	[MainMenuItem("Snapping", "Settings", "Select snapping modes")]
+	sealed class SnappingModule : MonoBehaviour, IUsesViewerScale, ISettingsMenuProvider
+	{
+		const float k_GroundSnappingMaxRayLength = 25f;
+		const float k_SurfaceSnappingMaxRayLength = 100f;
+
+		const float k_GroundHeight = 0f;
+
+		const float k_ManipulatorGroundSnapMin = 0.05f;
+		const float k_ManipulatorGroundSnapMax = 0.15f;
+		const float k_ManipulatorSurfaceSnapBreakDist = 0.1f;
+
+		const float k_DirectSurfaceSearchScale = 1.1f;
+		const float k_DirectSurfaceSnapBreakDist = 0.03f;
+		const float k_DirectGroundSnapMin = 0.03f;
+		const float k_DirectGroundSnapMax = 0.07f;
+
+		const float k_WidgetScale = 0.03f;
+
+		const string k_SnappingEnabled = "EditorVR.Snapping.Enabled";
```

Comment:
SnappingModule shouldn't be storing it's own settings directly in EditorPrefs, but you won't have the serialization system until the layout saving PR comes through.